sdk/aqueduct/decorator.py

```diff
@@ -451,6 +454,50 @@ def _update_operator_spec_with_resources(
         )
 
 
+def _update_operator_spec_with_image(
+    spec: OperatorSpec,
+    image: Optional[Dict[str, str]] = None,
+) -> None:
+    if image is not None:
+        if spec.engine_config is None or spec.engine_config.type is not RuntimeType.K8S:
+            engine_type = (
+                RuntimeType.AQUEDUCT if spec.engine_config is None else spec.engine_config.type
+            )
+            raise InvalidUserArgumentException(
+                "`image` is only compatible with Kubernetes engine. "
+                "Got engine type: %s. " % engine_type.value
+            )
+
+        if not isinstance(image, Dict) or any(not isinstance(k, str) for k in image):
+            raise InvalidUserArgumentException("`image` must be a dictionary with string keys.")
+
+        registry_name = image.get("registry_name")
+        url = image.get("url")
+
+        if registry_name is None:
+            raise InvalidUserArgumentException(
+                "`registry_name` must be specified when `image` is set."
+            )
+
+        connected_integrations = globals.__GLOBAL_API_CLIENT__.list_integrations()
+        if registry_name not in connected_integrations.keys():
+            raise InvalidUserArgumentException(
+                "Registry name `%s` is not one of the connected resources." % registry_name,
+            )
+
+        if url is None:
+            raise InvalidUserArgumentException("`url` must be specified when `image` is set.")
+
+        if len(url.split(":")) != 2:
```

Comment:
Reminder here to auto-add the `latest` tag when there is no tag provided by the user.